File: MATTHEW/leetcode/AIcoding/5.Qwen-3/c_solutions/661-image-smoother.c
#include <stdio.h>
#include <stdlib.h>

int** imageSmoother(int** img, int imgSize, int* imgColSize, int* returnSize, int** returnColumnSizes) {
    int rows = imgSize;
    int cols = imgColSize[0];
    int** result = (int**)malloc(rows * sizeof(int*));
    *returnColumnSizes = (int*)malloc(rows * sizeof(int));
    for (int i = 0; i < rows; i++) {
        result[i] = (int*)malloc(cols * sizeof(int));
        (*returnColumnSizes)[i] = cols;
    }
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            int sum = 0;
            int count = 0;
            for (int x = -1; x <= 1; x++) {
                for (int y = -1; y <= 1; y++) {
                    int ni = i + x;
                    int nj = j + y;
                    if (ni >= 0 && ni < rows && nj >= 0 && nj < cols) {
                        sum += img[ni][nj];
                        count++;
                    }
                }
            }
            result[i][j] = sum / count;
        }
    }
    *returnSize = rows;
    return result;
}

File: MATTHEW/leetcode/AIcoding/4.GPT-4o/go_solutions/661-image-smoother.go
func imageSmoother(M [][]int) [][]int {
    rows := len(M)
    if rows == 0 {
        return [][]int{}
    }
    cols := len(M[0])
    result := make([][]int, rows)
    for i := range result {
        result[i] = make([]int, cols)
    }

    directions := [][]int{
        {-1, -1}, {-1, 0}, {-1, 1},
        {0, -1},              {0, 0},  {0, 1},
        {1, -1}, {1, 0}, {1, 1},
    }

    for i := 0; i < rows; i++ {
        for j := 0; j < cols; j++ {
            sum := 0
            count := 0
            for _, d := range directions {
                ni, nj := i+d[0], j+d[1]
                if ni >= 0 && ni < rows && nj >= 0 && nj < cols {
                    sum += M[ni][nj]
                    count++
                }
            }
            result[i][j] = sum / count
        }
    }

    return result
}